### ISSUE TITLE
contrib/prometheus-exporter Add process metrics

### ISSUE THREAD

**Issue by SuperQ** [haproxy/haproxy#647]
## What should haproxy do differently? Which functionality do you think we should add?

In the contrib/prometheus-exporter, it would be useful to get process stats. At least on Linux, it's pretty easy to parse [`/proc/[pid]/stat`](http://man7.org/linux/man-pages/man5/proc.5.html) to get various fields.

The main ones we like to see in monitoring:
* `process_cpu_seconds_total` = (14) utime %lu + (15) stime %lu
* `process_resident_memory_bytes` = (24) RSS %ld

It would also be useful to gather the current open FDs so that it can be compared against `haproxy_process_max_fds`. I don't know of a more elegant way to get this other than getting a directory listing of `/proc/[pid]/fd/`.

Optionally, (23) vsize  %lu could be included, mostly useful on 32-bit systems.

## What are you trying to do?

Monitor the haproxy system resource use.

**Comment by SuperQ** [comment]
CC @roidelapluie

**Comment by roidelapluie** [comment]
This would be nice indeed. Not sure about the other platforms than linux TBH but those metrics would be really useful.

**Comment by git001** [comment]
This will only work when HAProxy isn't running in chroot or in containers.  
Why not using [process-exporter](https://github.com/ncabatoff/process-exporter) for such a metrics?

**Comment by SuperQ** [comment]
@git001 No, it will work just fine, each process can access stats via `/proc/self/stat`. We do this normally in [Prometheus client_golang](https://github.com/prometheus/client_golang/blob/master/prometheus/process_collector.go).

The only difficulty is when you have `nbproc` > 1. You have to gather the metrics from each worker process and label them accordingly. But for standard `nbproc` == 1 and `nbthread` setups, it works reasonably well.

**Comment by roidelapluie** [comment]
Note that instead of `/proc/pid` you can look at `/proc/self`
